pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -130,199 +146,244 @@ func TestAmbientIndex(t *testing.T) {
 
 	addPods("127.0.0.2", "name2", "sa1", map[string]string{"app": "a", "other": "label"}, nil)
 	addPods("127.0.0.3", "name3", "sa1", map[string]string{"app": "other"}, nil)
-	assertWorkloads("", "name1", "name2", "name3")
-	assertWorkloads("127.0.0.1", "name1")
-	assertWorkloads("127.0.0.2", "name2")
-	assert.Equal(t, controller.ambientIndex.Lookup("127.0.0.3"), []*model.WorkloadInfo{{
-		Workload: &workloadapi.Workload{
-			Name:              "name3",
-			Namespace:         "ns1",
-			Address:           netip.MustParseAddr("127.0.0.3").AsSlice(),
-			ServiceAccount:    "sa1",
-			Node:              "node1",
-			CanonicalName:     "other",
-			CanonicalRevision: "latest",
-			WorkloadType:      workloadapi.WorkloadType_POD,
-			WorkloadName:      "name3",
-			ClusterId:         "cluster0",
+	assertAddresses("", "name1", "name2", "name3")
+	assertAddresses("testnetwork/127.0.0.1", "name1")
+	assertAddresses("testnetwork/127.0.0.2", "name2")
+	assert.Equal(t, controller.ambientIndex.Lookup("testnetwork/127.0.0.3"), []*model.AddressInfo{
+		{
+			Address: &workloadapi.Address{
+				Type: &workloadapi.Address_Workload{
+					Workload: &workloadapi.Workload{
+						Name:              "name3",
+						Namespace:         "ns1",
+						Address:           netip.MustParseAddr("127.0.0.3").AsSlice(),
+						Network:           "testnetwork",
+						ServiceAccount:    "sa1",
+						Node:              "node1",
+						CanonicalName:     "other",
+						CanonicalRevision: "latest",
+						WorkloadType:      workloadapi.WorkloadType_POD,
+						WorkloadName:      "name3",
+						ClusterId:         "cluster0",
+						Status:            workloadapi.WorkloadStatus_HEALTHY,
+					},
+				},
+			},
 		},
-	}})
-	assertEvent("127.0.0.2")
-	assertEvent("127.0.0.3")
+	})
+	assertEvent("testnetwork/127.0.0.2")
+	assertEvent("testnetwork/127.0.0.3")
 
 	// Non-existent IP should have no response
-	assertWorkloads("10.0.0.1")
+	assertAddresses("testnetwork/10.0.0.1")
 	fx.Clear()
 
-	createService(controller, "svc1", "ns1",
+	addService("svc1", "ns1",
+		map[string]string{},
 		map[string]string{},
-		[]int32{80}, map[string]string{"app": "a"}, t)
-	// Service shouldn't change workload list
-	assertWorkloads("", "name1", "name2", "name3")
-	assertWorkloads("127.0.0.1", "name1")
+		[]int32{80}, map[string]string{"app": "a"}, "10.0.0.1")
+	// Services should appear with workloads
+	assertAddresses("", "name1", "name2", "name3", "svc1")
+	assertAddresses("testnetwork/127.0.0.1", "name1")
 	// Now we should be able to look up a VIP as well
-	assertWorkloads("10.0.0.1", "name1", "name2")
-	// We should get an event for the two *Pod* IPs impacted
-	assertEvent("127.0.0.1", "127.0.0.2")
+	assertAddresses("testnetwork/10.0.0.1", "name1", "name2", "svc1")
+	// We should get an event for the new Service and the two *Pod* IPs impacted
+	assertEvent("ns1/svc1.ns1.svc.cluster.local", "testnetwork/10.0.0.1", "testnetwork/127.0.0.1", "testnetwork/127.0.0.2")
```

Comment:
why ns1/svc1.ns1.svc.cluster.local? shouldn't it only be when waypoint changes, from updateEndpointsOnWaypointChange?

Reply:
Oh I guess maybe the normal non-ambient controller triggers it

Reply:
Now that waypoint is tied to service we may not even need updateEndpointsOnWaypointChange

Reply:
it apparently does not, surprisingly.  My [integration test](https://github.com/istio/istio/pull/43699) passes without.  I'm not sure if there were any other fixes addressed in https://github.com/istio/istio/pull/43692 that need to be tested

Reply:
and performed a manual sanity check that ingress gateway actually hits the waypoint too when one exists in front of productpage